stacker/blueprints/base.py

```diff
@@ -402,6 +406,44 @@ def render_template(self):
         version = hashlib.md5(rendered).hexdigest()[:8]
         return (version, rendered)
 
+    def parse_user_data(self, raw_user_data):
+        """Translate a userdata file to into the file contents.
+
+        It supports referencing template variables to create userdata
+        that's supplemented with information from the data, as commonly
+        required when creating EC2 userdata files. Automatically, encodes
+        the data file to base64 after it is processed.
+
+         Args:
+            raw_userdata (str of userdata):
+                str of the given userdata to be parsed
+
+        """
+        pattern = re.compile(r'{{([::|\w]+)}}')
+        user_data = []
+        start_index = 0
+        variables = self.get_variables()
+        parameters = self.get_cfn_parameters()
+
+        for match in pattern.finditer(raw_user_data):
```

Comment:
Lets go with the standard python `string.Template` method of templating: https://docs.python.org/3/library/string.html#template-strings

Also, please double check that the parameters aren't availabe in `get_variables` - seems strange, but worth double checking (@mhahn might know)